core/client/src/main/java/alluxio/client/file/policy/options/BlockLocationPolicyCreateOptions.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy.options;
+
+import alluxio.annotation.PublicApi;
+
+import com.google.common.base.Objects;
+
+/**
+ * Method options for creating a {@link BlockLocationPolicyCreateOptions}.
+ */
+@PublicApi
+public final class BlockLocationPolicyCreateOptions {
+  private String mLocationPolicyClassName;
+  private int mDeterministicHashPolicyNumShards;
+
+  /**
+   * @return the default {@link BlockLocationPolicyCreateOptions}
+   */
+  public static BlockLocationPolicyCreateOptions defaults() {
+    return new BlockLocationPolicyCreateOptions();
+  }
+
+  /**
+   * Creates a new instance with defaults.
+   */
+  private BlockLocationPolicyCreateOptions() {
+    mDeterministicHashPolicyNumShards = 1;
+  }
```

Comment:
Do you prefer to getting that from config (alluxio.ufs.read.location.policy) or just set it something like LocalFirstPolicy? Neither of them looks a perfect option for me.
1. CreateOptions is not just for ufs read. It is supposed to be a block policy that can be used for block write as well. So it looks strange to introducing ‘ufs.read’ specific property in this file.
2. If we set it to LocalFirstPolicy manually, it can become inconsistent with the default values of the site properties in the future which can hide bugs.